File: clone-index.py
# pip install beautifulsoup4
import os, json
from bs4 import BeautifulSoup

os.chdir("public")
with open("db.json") as f:
	data = json.load(f)
slug = []
judul = []
deskripsi = []
for x in data["postingan"]:
	slug.append(x["slug"])
	judul.append(x["judul"])
	deskripsi.append(x["deskripsi"])
for x in slug:
	os.system('cp index.html ' + x + '.html')
os.system("cp index.html 404.html")
os.system("cp index.html cari.html")

# data = open('todo.html')
# soup = BeautifulSoup(data.read())
# soup.find('title').string = 'TODO'
# soup.find('meta', {'name': 'description'})['content'] = 'TODO juga'
# print(soup)

for (n, x) in enumerate(slug):
	data = open(x + '.html')
	soup = BeautifulSoup(data.read(), 'html.parser')
	soup.find('title').string = judul[n]
	soup.find('meta', {'name': 'description'})['content'] = deskripsi[n]
	data.close()
	tulis = open(x + '.html', 'w')
	tulis.write(str(soup))
	tulis.close()

File: clone-index.js
process = require('process')
fs = require('fs')
util = require('util')
readFile = util.promisify(fs.readFile)
writeFile = util.promisify(fs.writeFile)
copyFile = util.promisify(fs.copyFile)
process.chdir('public')
buka = async () => {
	data = await readFile('db.json', 'utf8')
	data = JSON.parse(data)
	slug = []
	judul = []
	deskripsi = []
	for (x of data.postingan){
		slug.push(x.slug)
		judul.push(x.judul)
		deskripsi.push(x.deskripsi)
	}
	for (n in slug){
		baca = await readFile('index.html', 'utf8')
		baca = baca.replace(/(<title>)(.+)(<\/title>)/g, `$1${judul[n]}$3`)
		baca = baca.replace(/(<meta name="description" content=")(.+)(">)/g, `$1${deskripsi[n].replace(/"/g, '\\"')}$3`)
		writeFile(`${slug[n]}.html`, baca)
	}
	fs.copyFile('index.html', '404.html', () => {})
	fs.copyFile('index.html', 'cari.html', () => {})
}
buka()